src/discord-cluster-manager/bot.py

```diff
@@ -0,0 +1,125 @@
+import discord
+from discord import app_commands
+from discord.ext import commands
+import argparse
+from utils import setup_logging
+from cogs.misc_cog import MiscCog
+from datetime import datetime
+from consts import (
+    init_environment,
+    DISCORD_TOKEN,
+    DISCORD_DEBUG_TOKEN,
+    DISCORD_CLUSTER_STAGING_ID,
+    DISCORD_DEBUG_CLUSTER_STAGING_ID,
+)
+from cogs.modal_cog import ModalCog
+from cogs.github_cog import GitHubCog
+
+logger = setup_logging()
+
+
+class ClusterBot(commands.Bot):
+    def __init__(self, debug_mode=False):
+        intents = discord.Intents.default()
+        intents.message_content = True
+        super().__init__(intents=intents, command_prefix="!")
+        self.debug_mode = debug_mode
+
+        # Create the run group
+        self.run_group = app_commands.Group(
+            name="run", description="Run jobs on different platforms"
+        )
+        self.tree.add_command(self.run_group)
+
+    async def setup_hook(self):
+        logger.info(f"Syncing commands for staging guild {DISCORD_CLUSTER_STAGING_ID}")
+        try:
+            # Load cogs
+            await self.add_cog(ModalCog(self))
+            await self.add_cog(GitHubCog(self))
+            await self.add_cog(MiscCog(self))
```

Comment:
don't love the MiscCog name, can we be more explicit, this relates to my earlier point around what should or should not be a cog

EDIT: Maybe ManagerCog was what you're going for?